src/main/java/org/ice4j/ice/Component.java

```diff
@@ -943,9 +943,14 @@ public LocalCandidate findLocalCandidate(TransportAddress localAddress, LocalCan
             if (localCnd.getTransportAddress().equals(localAddress))
             {
                 if (base == null || base.equals(localCnd.getBase()))
+                {
                     return localCnd;
+                }
             }
         }
+        // In case the above loop failed to find a result because `base` was
+        // specified, fallback to the original behavior and return the first
+        // candidate matching `localAddress` regardless of `base`.
```

Comment:
Is this what we want? I was under the impression that the NPE we were seeing was coming about because had a candidate associated with the wrong base, so when we tried to send the nomination out through it it couldn't figure out how.

Reply:
No, we have the same address associated with two different local candidates (with different bases). Perhaps `localAddress` is not a good name -- it is used to match the candidate's address, which is not a local address in this case.